### DISCUSSION TITLE
Questions in relation to changing over from Windows to MacOS

### DISCUSSION THREAD

**Discussion by ZimTom** [zim-desktop-wiki/zim-desktop-wiki#2268] [General]
Hi all,
I am relatively new to ZIM (probable 5 months) and absolutely new to MacOS (couple of days really), but really love it. As the availablilty of ZIM as MacOS App was an argument for me to use it I now come across some questions that popped up while changing over from Windows to MacOS:

1. As I am new in the scene I can't really estimate how serious this might be, but there are statements/rumours about Rosetta disappearing short/mid term from MacOS and when I installed ZIM on my new Macbook Pro it requested to get Rosetta installed first. Are there plans for ZIM providing native Apple SoC support and how serious the more experienced Mac users see this issue ?

2. Having learned about the different approaches to GUI design in the Windows and MacOS world there is one thing that I saw already on the Windows edition, but becomes more prone under MacOS: In preferences I can change the font and adjust the size to be able to work comfortably in the pages themselves, but the page index on the left always keeps working with a standard font that makes it a bit hard to read/use on my Macbook. Is there a  way to adjust this as well or would it be an alternative to use the preference for the page font also for the index ?

That said, I haven't yet experienced any issues with the responsiveness of the GUI under Ventura 13.1

Thanks in advance for your feedback.

**Comment by dehesselle** [comment]
1. We have introduced a native Apple Silicon version 8 months ago. See our download page https://zim-wiki.org/downloads.html
2. This is probably something that can be changed with GTK's settings and theme files, but I haven't done anything like that myself.

You say you are not experiencing any issues with Ventura. I take it you've installed Zim with either MacPorts or Homebrew?

**Reply by ZimTom** [reply]
First of all I have to apologize for not coming back on this one. Must have missed it long time ago 🥴 ...

Today I am working with the latest ZIM version under Sonoma 14.5 and everything seems to be fine, apart from the fact that I still have no solution for Q2 in my original post of Dec 2022.

However In the meantime I found out that there should be a much simpler way to solve it, but I have difficulties to get it working.
In the ZIM Q&A poage I found the the section "Can I change the colors used for links, underline etc. ?" and I tried to work along the answer after reading the information in the ZIM documentation for Config Files.

I found the file "style.conf" in the zim.app package and I think I also have understood how to edit this file to make two modifications:

- Changing the style set by the underline button to "real" single underlining instead of the yellow background
- Changing the foreground color of links to the same green that is used for the headlines

These simple changes would solve the readability issues with the dark mode on the Mac.

The problem seems to be, that I didn't quite understand where to put the modified copy of the "style.conf" file so that ZIM actually sees and uses it on startup. I tried to create the mentioned path inside the zim.app package as well as under Libraries/Application support/... but ZIM doesn't seem to recognize the modified config file.

Can you help me with this, please ? I already went through the email channel in the Q&A page asking @jaap-karssenberg and he already gave me a hint that I tried, but he wasn's exactly sure how to do it in the Mac and I still can't make it work. It looks like I am still far, far away from understanding the structers behind MacOS compared to my Windows times ...

**Reply by introt** [reply]
@dehesselle could you give some Mac pointers, thanks!

**Reply by dehesselle** [reply]
Please attach your `style.conf` and point out what the intended changes are in case they're not self-explanatory or too subtle for me to notice when I try to get this to work.

**Comment by ZimTom** [comment]
Hi René,

Thanks a lot for coming back on this one.

I have attached the modified style.conf as requested.
The original file I took from the package zim.app in path /contents/Resources/share/zim.

Here is what I modified:

1. Change the format for the underline button tag to normal underlining 

[Tag mark]
underline = single
#background = yellow  

2. Change the format of links to the same colour as headlines

[Tag link]
foreground = #4e9a06
underline  = single

I hope with the help of the zim documentation I got the syntax right, but I am not sure where to put the modified style.conf  on the Mac. If I replace the original style.conf in the original location (path mentioned above) it seems to be ignored.

Best regards

Thomas

￼

**Reply by dehesselle** [reply]
General remark: you should never modify files inside an application bundle.

Now, to solve your issue, I can give you the answer straight away but that doesn't help improving the documentation.
- Did you read the macOS specific readme that is linked to on the download page? If not, do it now please.
- After having read the readme, do you know where to look?

If that doesn't help, I'll give you the answer you're looking for.

**Comment by ZimTom** [comment]
First of all thanks for the general remark.
After 40+ years in the Windows world I am still prone to SOE (Stupid Operator Errors) under MacOS. That said, in this case the modification I made didn’t seem to have any negative influence for now.
But I’ll remember this one.

Here’s what I did after receiving your mail:
  
Read again the FAQ about changing colors for links, underline etc.
Read again the Config Files page, especially the section there for MacOS
Read the Readme page behind the link in the MacOS download page (admittedly 1st time)

From (re)reading this, here is what this means to me:

According to the Config Files page as well as as the Readme page for MacOS the default path
for configuration files is /Library/Application Support/org.zim-wiki.Zim/  and
for data it is /Library/Application Support/org.zim-wiki.Zim/ share
So, to modify the styles I would expect to have to create a modified version of the file style.conf and store it in the path for for configuration files - rather than editing the file inside the package

When I tried to do this, the first thing I see is that this path - although it is the default path for configurations - didn’t exist.
I then created this path manually and stored my modified file styles.conf  in this path and restarted ZIM to see whether the file is recognized, but it looks like ZIM doesn’t see it or doesn’t accept it.

Regarding our intention about the documentation:

After knowing that there are slight differences in the procedure depending on the OS, I find the single line answer under FAQ confusing, especially because it doesn’t seem to match at all with the information in the Config Files page and your Readme for MacOS.

Hope that makes sense for you.

Best regards

Thomas

**Reply by dehesselle** [reply]
Thanks for sharing. You need to modify the file
```
~/Library/Application Support/org.zim-wiki.Zim/zim/style.conf
```
You seem to have missed the `~` indicating that this is in your user's home directory, not a system directory. The first time you start Zim, the `org.zim-wiki.Zim` folder is automatically created and populated for you. Make all your customizations there. Look around to see what files are in there.

edit: My mistake, the `style.conf` doesn't get created automatically. So the first step is
```
cp /Applications/Zim.app/Contents/Resources/share/zim/style.conf ~/Library/Application Support/org.zim-wiki.Zim/zim
```

**Reply by dehesselle** [reply]
Btw. SOE, nice one. I only knew PEBKAC. Problem Exists Between Keyboard And Chair. I blame my chair 😄 .

**Comment by ZimTom** [comment]
Thanks a lot for this hint. I indeed wasn’t aware about the secret syntax behind the „~“.
As I had a lot to do with IT-related topics in the last decades, but compared to people like you and Jaap probably am lightyears away from the perspective of a software developer, admittedly I am kind of prone to this kind of mishaps 🥴. I will try it out a.s.a.p. and I am sure that it will work straight away after your explanation.

Best regards and a nice (rest of) the evening.

Thomas

**Comment by ZimTom** [comment]
This is one for my collection as well 😁😁 !

Now’s the time to try !!! ZIM still thinks it is daytime, so I can watch live whether my stuff becomes readable after restart.

Thanks again and „May the Mac-Power be with you“

Thomas

**Comment by ZimTom** [comment]
I just tested it and here is the result ...

￼

So, the good news is that ZIM now reacts to the modification in style.conf, because with your help I am editing the right file now 🙂.
The changes for the [Tag link] work completely as desired, as it shows a link as single underlined in the green of the headlines, which is readable both in day and dark mode (the text doesn’t change color) 

But for some reason I can’t get rid of the yellow background color for [Tag mark] .
By removing the line background=yellow and inserting the line underline=single I achieved that the text is underlined correctly, but the yellow background is still there. And as there is not defined any foreground color the text changes correctly from black to white if MacOS changes to dark mode. However, due to the yellow background that also remains in dark mode the the white text becomes unreadable.

I now have double-checked the modifications in style.conf (the right one 😉) and have removed any modifications I made accidentally in the system-related path and the package itself, but I still can’t get rid of this yellow color.
Do you have any idea why this happens ?

Best regards

Thomas

**Comment by ZimTom** [comment]
Problem solved !!!

If in the section [Tag mark] the line background=none is added, this seems to override the background setting coming from somewhere other than the style.conf and now I get correct display of underlined text black on white in day mode and white on black in dark mode.

Because you were thinking about the documentation, maybe it would be a good idea to describe the style.conf options and syntax in a bit more detail some time in the future.

Have a good night and I hope you don’t dream in underlined text now 😉.

Regards Thomas

**Comment by dehesselle** [comment]
Seems resolved, closing.
